swift/Samples/SplitScreenContainer/Sources/SplitScreenContainerViewController.swift

```diff
@@ -0,0 +1,103 @@
+    
+    private var leftContentViewController: ScreenViewController<LeftScreenType>? = nil
+    private lazy var leftContainerView: UIView = .init()
+
+    private lazy var separatorView: UIView = .init()
+
+    private var rightContentViewController: ScreenViewController<RightScreenType>? = nil
+    private lazy var rightContainerView: UIView = .init()
+    
+    private var currentRatio: CGFloat {
+        didSet {
+            guard oldValue != currentRatio else {
+                return
+            }
+            
+            UIView.animate(withDuration: 0.25) {
+                self.view.setNeedsLayout()
+                self.view.layoutIfNeeded()
+            }
+        }
+    }
+
+    required init(screen: ContainerScreen, viewRegistry: ViewRegistry) {
+        currentRatio = screen.ratio
+        
+        super.init(screen: screen, viewRegistry: viewRegistry)
+    }
+
+    override public func screenDidChange(from previousScreen: ContainerScreen) {
+        update(with: screen)
+    }
+
+    private func update(with screen: ContainerScreen) {
+        separatorView.backgroundColor = screen.separatorColor
+        
+        leftContentViewController?.update(screen: screen.leftScreen)
+        rightContentViewController?.update(screen: screen.rightScreen)
+        
+        //Intentional force of layout pass after updating the child view controllers
+        view.layoutIfNeeded()
+        
+        //Update the current ratio in an animated fashion if needed
+        currentRatio = screen.ratio
+    }
+
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.addSubview(leftContainerView)
+        view.addSubview(separatorView)
+        view.addSubview(rightContainerView)
+        
+        self.leftContentViewController = embed(screen.leftScreen, in: leftContainerView)
+        self.rightContentViewController = embed(screen.rightScreen, in: rightContainerView)
+        update(with: screen)
+    }
+    
+    public override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        
+        let distance = view.bounds.width * currentRatio
+        
+        let (firstSlice, rightRect) = view.bounds.divided(atDistance: distance, from: .minXEdge)
+        
+        let (leftRect, separatorRect) = firstSlice.divided(atDistance: distance - 1, from: .minXEdge)
+        
+        leftContainerView.frame = leftRect
+        separatorView.frame = separatorRect
+        rightContainerView.frame = rightRect
+        leftContentViewController?.view.frame = leftRect
+        rightContentViewController?.view.frame = CGRect(x: 0, y: 0, width: rightRect.width, height: rightRect.height)
+    }
+    
+    private func embed<ScreenType: Screen>(_ screen: ScreenType, in containerView: UIView) -> ScreenViewController<ScreenType> {
+        let viewController = viewRegistry.provideView(for: screen)
+        
+        addChild(viewController)
+        containerView.addSubview(viewController.view)
```

Comment:
Do we need to resize the `viewController.view`'s frame to match `containerView.bounds`?

Reply:
It's not really necessary as it's done as part of `viewDidLayoutSubviews`, which happens after the two calls to the embed function (one for left and one for right).

Reply:
In person conversation led us to agree to extract a "ContainerView" that always set it's contentView's frame to match superview's bounds in layoutSubviews.

Reply:
Resolved via https://github.com/square/workflow/pull/925/commits/ae830a68008b85f7f7cc2e7003c811213ab70043